### PR TITLE
[MPS] Allow `float16` input to float32 `LayerNorm` (#96430)

### PR DESCRIPTION
**Pull request by malfet**
Only for forward pass. This also fixes a regular float16 normalization path, i.e. something like:
```
torch.nn.LayerNorm((16,), elementwise_affine=True).to("mps", dtype=torch.float16)(torch.randn(1, 2, 16).to("mps", dtype=torch.float16))
```

Subset of https://github.com/pytorch/pytorch/pull/96208

Create constant with scalar using `input_mps_dtype` and use
`reciprocalWithTensor` instead of `divisionWithPrimaryTensor:1.0
secondaryTensor:`

Fixes https://github.com/pytorch/pytorch/issues/96113

Pull Request resolved: https://github.com/pytorch/pytorch/pull/96430
Approved by: https://github.com/kulinseth

(cherry picked from commit 075a49442dc813537075063fd2efb779a50ebe30)

### CODE REVIEW

**Comment by kulinseth** [comment]
@malfet , as per discussion, do we still want this PR in 2.0.1 ?

**Comment by kulinseth** [comment]
@malfet , i don't think we need this PR as it already merged..
